crates/core/src/models/felt.rs

```diff
@@ -90,3 +90,106 @@ impl From<Felt252Wrapper> for U256 {
         U256::from_be_bytes(felt.to_bytes_be())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use hex::FromHex;
+
+    use super::*;
+
+    const MAX_ADDRESS: &str = "ffffffffffffffffffffffffffffffffffffffff";
+    const OVERFLOW_ADDRESS: &str = "010000000000000000000000000000000000000000";
+
+    const MAX_FELT: &str = "0800000000000011000000000000000000000000000000000000000000000000";
+    const OVERFLOW_FELT: &str = "0800000000000011000000000000000000000000000000000000000000000001";
+
+    #[test]
+    fn test_address_try_from_felt_should_pass() {
+        // Given
+        let address: Felt252Wrapper = FieldElement::from_hex_be(MAX_ADDRESS).unwrap().into();
+
+        // When
+        let address = Address::try_from(address).unwrap();
+
+        // Then
+        let expected_address = <[u8; 20]>::from_hex(MAX_ADDRESS).unwrap();
+        assert_eq!(expected_address, address.0);
+    }
+
+    #[test]
+    fn test_address_try_from_felt_should_fail() {
```

Comment:
see https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic

we should either use the `#[should_panic]` macro or returns a `Result`